### DISCUSSION TITLE
Categories with Custom Listing pages

### DISCUSSION THREAD

**Discussion by phdjsep** [quarto-dev/quarto-cli#6974] [Q&A] [answered]
### Description

Hi, I have a custom listing page that's exactly like that of the Quarto Gallery implementation, but can't seem to determine how to enable "categories". Similar to a post, I'd like to have multiple categories applied to a given listing item. 

Example:
```
- name: First Item
  href: https://www.quarto.org
  categories: [website, personal site]
- name: Second Item
  href: https://www.rstudio.org
  categories: [website, portfolio]
```

I'm open to other variations of implementation, but as of now, enabling categories just lists the high-level "parent" category.

**Comment by mcanouil** [comment]
Your snippet is not enough as it only show `contents` whilst the categories, it's documented at <https://quarto.org/docs/websites/website-listings.html#categories>.
If by custom, you mean you are using an EJS template, then you have to make categories appear in it youself, especially if you are not sharing a fully reproducible example.

**Comment by phdjsep** [comment]
Apologies for the lack of detail @mcanouil 

I've created a reprex site to make it easier: https://github.com/phdjsep/custom-listing-reprex

In my `index.qmd` I have the following `categories: true` in my YAML heading to enable the category sidebar.

```
listing:
  - id: gallery
    categories: true
    template: gallery.ejs
    contents: gallery.yml
    sort-ui: false
    filter-ui: false
    page-size: 5
```

At the moment, the category sidebar will only show `All (1)`, but I'm interested in how I need to either 1) format the `gallery.yaml` to  make the category sidebar display the other categories or 2) Create/configure something else to accommodate the existing `gallery.yaml` format.

I'm trying to achieve something like what can be found on the [hugo theme site](https://themes.gohugo.io/). 

Hopefully that's a bit more clear.

**Reply by mcanouil** [reply, marked as answer]
I can't do it for you, but here you have to also define an HTML partial for where categories should go.

You can take the following as example/inspiration:
- https://github.com/andrewheiss/ath-quarto/blob/main/research/index.qmd
- https://github.com/andrewheiss/ath-quarto/tree/main/html/research
<img width="1504" alt="image" src="https://github.com/quarto-dev/quarto-cli/assets/8896044/fd9b4462-9f0b-48f9-a0aa-9377d8d98c16">

**Comment by phdjsep** [comment]
Merci! I just needed some direction. Appreciate the help.
